tools/js-tools/tsconfig.tsc-esm.json

```diff
@@ -0,0 +1,12 @@
+{
```

Comment:
explicitly transpiles to esm rather than leave it to decide by the package, like in Jetpack components:

https://github.com/Automattic/jetpack/blob/899fc51f4d4eab20af81e0a40fb25af49d632c87/projects/js-packages/components/package.json#L20

With this config, we explicitly set the type to esm to enable dual build with common js

Reply:
This file is not needed if you use [this approach](https://github.com/Automattic/jetpack/pull/45051#pullrequestreview-3182904107)